src/Symfony/Component/Console/Helper/Table.php

```diff
@@ -138,6 +144,56 @@ public function getStyle()
         return $this->style;
     }
 
+    /**
+     * Sets table style.
+     *
+     * @param int               $columnIndex Column index
+     * @param TableStyle|string $name        The style name or a TableStyle instance
+     *
+     * @return Table
+     */
+    public function setColumnStyle($columnIndex, $name)
+    {
+        $isIndexValid = false;
+        if (is_int($columnIndex)) {
+            $isIndexValid = true;
+        } elseif (is_string($columnIndex)) {
+            $isIndexValid = ctype_digit($columnIndex);
+        }
+
+        if (!$isIndexValid) {
+            throw new \InvalidArgumentException(sprintf('Invalid column index: "%s".', $columnIndex));
+        }
+
+        $columnIndex = intval($columnIndex);
+
+        if ($name instanceof TableStyle) {
+            $this->columnStyles[$columnIndex] = $name;
+        } elseif (isset(self::$styles[$name])) {
+            $this->columnStyles[$columnIndex] = self::$styles[$name];
+        } else {
+            throw new \InvalidArgumentException(sprintf('Style "%s" is not defined.', $name));
+        }
+
+        return $this;
+    }
+
+    /**
+     * Gets the current style for specified column.
```

Comment:
I would expand the phpdoc with a long description explaining that the global table style is returned for columns which don't have their own style